cli/src/cmd/build.rs

```diff
@@ -251,6 +260,25 @@ impl BuildArgs {
     }
 }
 
+// Make this args a `Figment` so that it can be merged into the `Config`
+impl Provider for BuildArgs {
+    fn metadata(&self) -> Metadata {
+        Metadata::named("Build Args Provider")
+    }
+
+    fn data(&self) -> Result<Map<Profile, Dict>, Error> {
+        let mut dict = Dict::new();
+
+        if let Some(ref src) = self.contracts {
+            dict.insert("src".to_string(), Value::from(format!("{}", src.display())));
+        }
+
+        // TODO remaining configs
+
+        Ok(Map::from([(Config::selected_profile(), dict)]))
+    }
```

Comment:
Is the TODO for this PR, or for a follow up?

FWIW me doing `forge install` does not populate the config file with the new remapping, and `forge remappings` doesn't read any remappings from the config file. I guess that's intended right?

Reply:
still todo for this, is close now.

forge install probably should add them by default, that should be possible.

hmm, perhaps we let `forge remappings` always autodetect everything by scanning the libs, should be more useful then reading the config I guess